packages/near-snap/src/lib/near-snap.ts

```
@@ -0,0 +1,110 @@
+import {
+  WalletModuleFactory,
+  WalletBehaviourFactory,
+  InjectedWallet,
+} from "@near-wallet-selector/core";
+import { isMobile } from "is-mobile";
+import { enable, getSnapAccounts, signTransactions } from "./methods";
+import { isMetaMaskAvailable } from "./utils";
+
+const isDev = false;
+
+const NearSnapWallet: WalletBehaviourFactory<InjectedWallet> = async ({
+  options,
+  provider,
+  store,
+}) => {
+  const network = options.network.networkId as "testnet" | "mainnet";
+
+  return {
+    async signIn() {
+      // Sign in to My Wallet for access to account(s).
+      await enable(isDev);
+
+      return await getSnapAccounts(isDev, network);
+    },
+
+    async signOut() {
+      // Nothing to do here!!
+    },
+
+    async getAccounts() {
+      return await getSnapAccounts(isDev, network);
+    },
+
+    // eslint-disable-next-line @typescript-eslint/no-empty-function
+    async verifyOwner() {},
```

Can we throw an error if this method is not implemented in `Near Snap`?

```ts
async verifyOwner() {
    logger.log("NearSnap:verifyOwner", { message });
    
    throw new Error(`Method not supported by ${metadata.name}`);
}
```

---

Can you please take a look at this again?